crypto/rand/rand_unix.c

```diff
@@ -486,39 +574,42 @@ size_t rand_pool_acquire_entropy(RAND_POOL *pool)
 #   endif
 
 #   if defined(OPENSSL_RAND_SEED_DEVRANDOM)
-    bytes_needed = rand_pool_bytes_needed(pool, 1 /*entropy_factor*/);
-    {
-        size_t i;
-
-        for (i = 0; bytes_needed > 0 && i < OSSL_NELEM(random_device_paths); i++) {
-            ssize_t bytes = 0;
-            /* Maximum allowed number of consecutive unsuccessful attempts */
-            int attempts = 3;
-            const int fd = get_random_device(i);
-
-            if (fd == -1)
-                continue;
-
-            while (bytes_needed != 0 && attempts-- > 0) {
-                buffer = rand_pool_add_begin(pool, bytes_needed);
-                bytes = read(fd, buffer, bytes_needed);
-
-                if (bytes > 0) {
-                    rand_pool_add_end(pool, bytes, 8 * bytes);
-                    bytes_needed -= bytes;
-                    attempts = 3; /* reset counter after successful attempt */
-                } else if (bytes < 0 && errno != EINTR) {
-                    break;
+    if (wait_random_seeded()) {
+        bytes_needed = rand_pool_bytes_needed(pool, 1 /*entropy_factor*/);
+        {
+            size_t i;
```

Comment:
you could move this decl betore bytes_needed, and avoid changing the indentation in here.

Reply:
Oops, I had this condition somewhat rearranged and missed fixing it.